go/timer/timer.go

```diff
@@ -87,13 +87,23 @@ func (tm *Timer) Start(keephouse func()) {
 }
 
 func (tm *Timer) run(keephouse func()) {
+	var timer *time.Timer
+	stopTimer := func() {
+		if timer != nil {
+			timer.Stop()
+		}
+		timer = nil
+	}
+	defer stopTimer()
 	for {
+		stopTimer()
 		var ch <-chan time.Time
 		interval := tm.interval.Get()
 		if interval <= 0 {
```

Comment:
While we're here, we can delete this section and change the condition to `interval > 0`, because ch is already nil.